src/main/java/com/allocadia/carbonite/PersistedObjectCache.java

```diff
@@ -20,4 +24,14 @@
     public <T, I> T getObjectById(Class<T> clazz, I id, Function<I, T> fetcher) {
         return getInstanceCache(clazz, id).computeIfAbsent(Objects.requireNonNull(id), fetcher.andThen(Objects::requireNonNull));
     }
+
+    public <T extends PersistedObject> void addDirty(T po) {
```

Comment:
I guess I was combining two separate things:

1. Somehow we should expose `getObjectById` like JDO does, probably along with `save` and `delete`. Which object will expose that? And this `addDirty` should *not* be exposed despite the implementation currently being beside `getObjectById`
2. I think the logic in `PersistanceAspect#markFieldDirty` should move here so that a) the aspect has no logic in it b) both the dirty field and dirty object logic is in one spot